api/v1/storagecluster_types.go

```diff
@@ -137,11 +140,41 @@ type ManageCephObjectStoreUsers struct {
 	ReconcileStrategy string `json:"reconcileStrategy,omitempty"`
 }
 
+// ExternalStorageKind specifies a kind of the external storage
+type ExternalStorageKind string
+
+const (
+	// KindOcs specifies a "ocs" kind of the external storage
+	KindOcs ExternalStorageKind = "ocs"
+
+	// KindRhcs specifies a "rhcs" kind of the external storage
+	KindRhcs ExternalStorageKind = "rhcs"
+)
+
 // ExternalStorageClusterSpec defines the spec of the external Storage Cluster
-// to be connected to the local cluster
+// to be connected to the external cluster
 type ExternalStorageClusterSpec struct {
-	// +optional
+	//+kubebuilder:default:=true
+	// Enable enables the external mode
 	Enable bool `json:"enable,omitempty"`
+
+	//+kubebuilder:default:=rhcs
+	// StorageProviderKind describes the kind of the external storage
+	StorageProviderKind ExternalStorageKind `json:"storageProviderKind,omitempty"`
+
+	// ConnectionString describes the blob of provider
+	ConnectionString string `json:"connectionString,omitempty"`
+
+	//+kubebuilder:default:="1Ti"
+	// RequestedCapacity defines the requested storage on the provider
+	RequestedCapacity resource.Quantity `json:"requestedCapacity,omitempty"`
+}
+
+// ExternalStorageClusterStatus defines the status of the external Storage Cluster
+// to be connected to the external cluster
+type ExternalStorageClusterStatus struct {
+	// ActualCapacity defines the actual storage provided by the provider
+	ActualCapacity resource.Quantity `json:"actualCapacity,omitempty"`
```

Comment:
Does this require some sort of Succeded/Failed? status as well ?

Reply:
not really, It should be handled the way it is currently handled in the external mode IMO